test/framework/src/main/java/org/elasticsearch/test/junit/listeners/LoggingListener.java

```diff
@@ -89,22 +99,35 @@ private static Logger resolveLogger(String loggerName) {
      * @param testLogging the test logging annotation to apply
      * @return the existing logging levels
      */
-    private Map<String, String> processTestLogging(final TestLogging testLogging) {
-        final Map<String, String> map = getLoggersAndLevelsFromAnnotation(testLogging);
-
-        if (map == null) {
-            return Collections.emptyMap();
+    private Map<String, String> processTestLogging(final TestLogging testLogging, final TestIssueLogging testIssueLogging) {
+        final Map<String, String> testLoggingMap = getLoggersAndLevelsFromAnnotation(testLogging);
+        final Map<String, String> testIssueLoggingMap = getLoggersAndLevelsFromAnnotation(testIssueLogging);
+
+        final Set<String> testLoggingKeys = new HashSet<>(testLoggingMap.keySet());
+        testLoggingKeys.retainAll(testIssueLoggingMap.keySet());
+        if (testLoggingKeys.isEmpty() == false) {
+            throw new IllegalArgumentException("found intersection " + testLoggingKeys + " between TestLogging and TestIssueLogging");
         }
 
-        // obtain the existing logging levels so that we can restore them at the end of the test; we have to do this separately from setting
-        // the logging levels so that setting foo does not impact the logging level for foo.bar when we check the existing logging level for
-        // for.bar
+        /*
+         * Use a sorted set so that we apply a parent logger before its children thus not overwriting the child setting when processing the
```

Comment:
Is this comment relevant here any more ?  If so I don't think `Collectors.toMap` has any guarantees on ordering

Reply:
The collector won't, but the returned map is passed to TreeMap, which will be sorted.

Reply:
The sorting is indeed still necessary, otherwise when we set the logger level for `foo` it could override the logger level for `foo.bar` (we have to set the child logger level last, if both are annotated). As @rjernst points out, that's why pass the collected map to a sorted map implementation.